### PR TITLE
[32017] Marked Directory, Downloadable, Eav and Elasticsearch interfaces as API

### LINKED ISSUE #32017: Mark Directory, Downloadable, Eav and Elasticsearch interfaces as API
The following interfaces should be marked as API in 2.4-develop branch
1. app/code/Magento/Directory/Model/Country/Postcode/ConfigInterface.php
1. app/code/Magento/Directory/Model/Country/Postcode/ValidatorInterface.php
1. app/code/Magento/Downloadable/Api/DomainManagerInterface.php
1. app/code/Magento/Downloadable/Pricing/Price/LinkPriceInterface.php
1. app/code/Magento/Eav/Model/Entity/Attribute/AttributeInterface.php
1. app/code/Magento/Eav/Model/Entity/Attribute/Backend/BackendInterface.php
1. app/code/Magento/Eav/Model/Entity/Attribute/Source/SpecificSourceInterface.php
1. app/code/Magento/Eav/Model/Entity/Attribute/UniqueValidationInterface.php
1. app/code/Magento/Eav/Model/Entity/AttributeLoaderInterface.php
1. app/code/Magento/Eav/Model/Entity/Setup/PropertyMapperInterface.php
1. app/code/Magento/Eav/Model/ResourceModel/Attribute/DefaultEntityAttributes/ProviderInterface.php
1. app/code/Magento/Elasticsearch/Model/Adapter/FieldMapper/Product/FieldProvider/FieldIndex/ConverterInterface.php
1. app/code/Magento/Elasticsearch/Model/Adapter/FieldMapper/Product/FieldProvider/FieldIndex/ResolverInterface.php
1. app/code/Magento/Elasticsearch/Model/Adapter/FieldMapper/Product/FieldProvider/FieldName/ResolverInterface.php
1. app/code/Magento/Elasticsearch/Model/Adapter/FieldMapper/Product/FieldProvider/FieldType/ResolverInterface.php
1. app/code/Magento/Elasticsearch/Model/Adapter/FieldMapper/Product/FieldProviderInterface.php
1. app/code/Magento/Elasticsearch/Model/Adapter/FieldsMappingPreprocessorInterface.php
1. app/code/Magento/Elasticsearch/SearchAdapter/Query/ValueTransformerInterface.php
1. app/code/Magento/Elasticsearch/SearchAdapter/QueryAwareInterface.php

### PR DESCRIPTION
**Pull request by Usik2203**
### Description (*)

This PR fixes https://github.com/magento/magento2/issues/32017
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32017

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Usik2203** [comment]
@magento run all tests

File: app/code/Magento/Eav/Model/ResourceModel/Attribute/DefaultEntityAttributes/ProviderInterface.php [hunk 1]
```diff
@@ -3,11 +3,15 @@
...
  */
+declare(strict_types=1);
+
 namespace Magento\Eav\Model\ResourceModel\Attribute\DefaultEntityAttributes;
 
 /**
  * Interface \Magento\Eav\Model\ResourceModel\Attribute\DefaultEntityAttributes\ProviderInterface
  *
+ * @api
+ *
```
Comment (coderimus) [inline app/code/Magento/Eav/Model/ResourceModel/Attribute/DefaultEntityAttributes/ProviderInterface.php]: Please, remove an empty line after the @api

File: app/code/Magento/Eav/Model/Entity/Setup/PropertyMapperInterface.php [hunk 2]
```diff
@@ -5,11 +5,15 @@
...
  */
+declare(strict_types=1);
+
 namespace Magento\Eav\Model\Entity\Setup;
 
 /**
  * Interface \Magento\Eav\Model\Entity\Setup\PropertyMapperInterface
  *
+ * @api
+ *
```
Comment (coderimus) [inline app/code/Magento/Eav/Model/Entity/Setup/PropertyMapperInterface.php]: Please, remove an empty line after the @api

File: app/code/Magento/Eav/Model/Entity/Attribute/Backend/BackendInterface.php [hunk 3]
```diff
@@ -11,6 +13,8 @@
  * Backend is responsible for saving the values of the attribute
  * and performing pre and post actions
  *
+ * @api
+ *
```
Comment (coderimus) [inline app/code/Magento/Eav/Model/Entity/Attribute/Backend/BackendInterface.php]: Please, remove an empty line after the @api

File: app/code/Magento/Eav/Model/Entity/Attribute/AttributeInterface.php [hunk 4]
```diff
@@ -3,11 +3,15 @@
...
  */
+declare(strict_types=1);
+
 namespace Magento\Eav\Model\Entity\Attribute;
 
 /**
  * Interface \Magento\Eav\Model\Entity\Attribute\AttributeInterface
  *
+ * @api
+ *
```
Comment (coderimus) [inline app/code/Magento/Eav/Model/Entity/Attribute/AttributeInterface.php]: Please, remove an empty line after the @api

**Review by coderimus — CHANGES_REQUESTED** [review]
Hi @Usik2203 thank you for the adjustments. Please, check my comments about the empty lines. Everything else good 👍

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by Usik2203** [comment]
Hi @coderimus 
I have applied your recommendations .
Thanks !

**Review by coderimus — APPROVED** [review]

**Comment by coderimus** [comment]
@Usik2203 thank you for the adjustments 👍

**Comment by magento-engcom-team** [comment]
Hi @coderimus, thank you for the review.
[ENGCOM-8763](https://jira.corp.magento.com/browse/ENGCOM-8763) has been created to process this Pull Request

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
Closed in favour of https://github.com/magento/magento2/pull/32726

**Pull request closed without merging**